File: artifacts/avl-tree/manual_translation/src/avltree.rs
use crate::*;


pub struct AvlTree
{
    pub compare: fn(Option<&Mydata>, Option<&Mydata>) -> i32,//Option<Box<dyn Fn(&dyn std::any::Any, &dyn std::any::Any) -> i32>>,
    pub print: fn(Option<&Mydata>),//Option<Box<dyn Fn(&dyn std::any::Any)>>,
    pub destroy: fn(Option<Box<Mydata>>),//Option<Box<dyn Fn(&dyn std::any::Any)>>,
    pub root: Option<Box<AvlNode>>,
    pub nil: Option<Box<AvlNode>>,
    pub min: Option<Box<AvlNode>>,
}

File: artifacts/avl-tree/decompose/src/avltree.c
typedef struct 
{
  int (*compare)(const void *, const void *);
  void (*print)(void *);
  void (*destroy)(void *);
  avlnode root;
  avlnode nil;
  avlnode *min;
} avltree
/*** DEPENDENCIES:
typedef struct avlnode
{
  struct avlnode *left;
  struct avlnode *right;
  struct avlnode *parent;
  char bf;
  void *data;
} avlnode
----------------------------
***/


